dds/DCPS/DataReaderImpl.cpp

```diff
@@ -3384,57 +3379,6 @@ DataReaderImpl::get_ice_endpoint()
   return TransportClient::get_ice_endpoint();
 }
 
-DDS::ReturnCode_t DataReaderImpl::setup_deserialization()
```

Comment:
I think I see what you mean by part of this being already being done in match. However the checks being done in `dds_demarshal` and `lookup_instance` are more direct as it doesn't rely on what QoS is saying. So it's not going to get triggered under normal conditions, but I think it would be useful as a final internal check. For example if there was a bug where QoS and the encoding didn't line up anymore, this could help reveal that. Even if that wasn't the case or we wanted to get rid of those checks anyways there are things being done here that I'm not certain are being done anywhere else. I used the ["Setup successfully with the following data representation" message in the issue that prompted these changes](https://github.com/objectcomputing/OpenDDS/issues/2946#issuecomment-906529994).

So I don't think we should remove any of this unless it's actually causing an issue.

Reply:
Well actually the encoding comes from the QoS, so that's a bad example unless something was really wrong, like one side turned the QoS into the wrong encoding. I guess the more possible scenario would if something was wrong with matching.

Reply:
Okay. I readded the removed functions. It still checks `cdr_encapsulation()` but this is used to determine whether the default for a reader should be unaligned cdr (blank QoS, non rtps_udp transport). The reason all this was removed in the first place was it was forcing unaligned, regardless of the QoS, this solution should be more elegant.